bundle/edu.gemini.wdba.xmlrpc.server/src/main/java/edu/gemini/wdba/tcc/TccFieldConfig.java

```diff
@@ -53,21 +53,16 @@ public boolean build() throws WdbaGlueException {
         putParameter(TccNames.GUIDE_GROUP, getPrimaryGuideGroupName());
 
         // Add a rotator parameter and then create the config
-        // Note: If the config name contains "Fixed" in some way, it's assumed to be a Fixed rotator config
-        RotatorConfig rc = new RotatorConfig(_oe);
+        final RotatorConfig rc = new RotatorConfig(_oe);
         if (rc.build()) {
-            String rotConfigName = rc.getConfigName();
-            if (rotConfigName.equals(TccNames.ALTAIR_FIXED)) {
-                putParameter(TccNames.ROTATOR, rotConfigName);
-            } else
             // If the instrument adds something, create a rotator config and add it in
-            if (rotConfigName.contains(TccNames.FIXED)) {
-                putParameter(TccNames.ROTATOR, rotConfigName);
-                add(rc);
-            } else {
-                putParameter(TccNames.POSANGLE, rotConfigName);
-                add(rc);
-            }
+            ImOption.apply(rc.attributeValue(TYPE)).foreach(t -> {
+                final String name = rc.getConfigName();
+                putParameter(t, name);
```

Comment:
We always should add a parameter whose name is the same as the rotator config type (which is `posAngle` or `rotator`), so instead of checking the rotator config name and inferring the type from that I decided to so something radical and just use the type as the name directly.  This was then monkey tested and unit tested.